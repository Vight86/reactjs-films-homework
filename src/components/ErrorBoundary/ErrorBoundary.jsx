import React, { Component } from 'react';
import propTypes from 'prop-types';
import styles from './ErrorBoundary.scss';

export default class ErrorBoundary extends Component {
  constructor(props) {
    super(props);
    this.state = { error: null };
  }

  componentDidCatch(error) {
    this.setState({
      error,
    });
  }

  render() {
    const { error } = this.state;
    if (error) {
      return (
        <div className={styles.errorWrapper}>
          <h1 className={styles.errorTitle}>Error</h1>
          <p className={styles.errorText}>Please try to reload this page</p>
        </div>
      );
    }

    const { children } = this.props;

    return children;
  }
}

ErrorBoundary.propTypes = {
  children: propTypes.element.isRequired,
};
