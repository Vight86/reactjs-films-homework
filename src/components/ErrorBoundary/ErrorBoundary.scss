@import '../../utils/variables';

.errorWrapper {
  width: 50%;
  margin: 2em auto;
  text-align: center;
}

.errorTitle {
  margin-bottom: 1em;
  color: $color-secondary;
  font-size: 3em;
  line-height: normal;
}

.errorText {
  color: $text-tertiary;
  font-size: 1.8em;
  line-height: normal;
}
