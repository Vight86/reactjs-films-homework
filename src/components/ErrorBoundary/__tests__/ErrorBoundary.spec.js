import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import ErrorBoundary from '../index';

const MockReactElem = () => <div>Test</div>;

let renderer;
let result;

beforeEach(() => {
  renderer = new ShallowRenderer();
  renderer.render(
    <ErrorBoundary>
      <MockReactElem />
    </ErrorBoundary>,
  );
  result = renderer.getRenderOutput();
});

afterEach(() => {
  renderer = null;
  result = null;
});

describe('test ErrorBoundary component', () => {
  it('renders component correctly', () => {
    expect(result).toMatchSnapshot();
  });

  it('component catches errors correctly', () => {
    const errorBoundary = new ErrorBoundary();
    const mockSetState = jest.fn();
    errorBoundary.setState = mockSetState;
    errorBoundary.componentDidCatch();
    expect(mockSetState).toHaveBeenCalled();
  });

  it('component renders correctly after errors are cought', () => {
    const errorBoundary = new ErrorBoundary();
    errorBoundary.state.error = true;
    expect(errorBoundary.render()).toMatchSnapshot();
  });
});
