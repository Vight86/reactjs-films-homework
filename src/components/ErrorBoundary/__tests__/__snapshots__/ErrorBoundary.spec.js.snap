// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test ErrorBoundary component component renders correctly after errors are cought 1`] = `
<div
  className="errorWrapper"
>
  <h1
    className="errorTitle"
  >
    Error
  </h1>
  <p
    className="errorText"
  >
    Please try to reload this page
  </p>
</div>
`;

exports[`test ErrorBoundary component renders component correctly 1`] = `<MockReactElem />`;
